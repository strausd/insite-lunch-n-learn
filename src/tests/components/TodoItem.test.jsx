import React from 'react';
import { shallow } from 'enzyme';

import TodoItemConnected, { TodoItem } from '../../components/TodoItem';
import { todos } from '../mockdata/todos';


let wrapper, deleteTodoSpy, toggleCompletedSpy;

beforeAll(() => {
    deleteTodoSpy = jest.fn();
    toggleCompletedSpy = jest.fn();
    wrapper = shallow(<TodoItem todo={todos[0]} deleteTodo={deleteTodoSpy} toggleCompleted={toggleCompletedSpy}/>);
});

test('should render TodoItem component properly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should call handleDeleteTodo on delete button click', () => {
    wrapper.find('button').at(0).simulate('click');
    expect(deleteTodoSpy).toHaveBeenLastCalledWith(todos[0].id);
});

test('should call handleToggleComplete on toggle complete button click', () => {
    wrapper.find('button').at(1).simulate('click');
    expect(toggleCompletedSpy).toHaveBeenLastCalledWith(todos[0].id);
});

test('should have not-complete class on todo paragraph tag', () => {
    expect(wrapper.find('p').hasClass('not-complete')).toEqual(true);
    expect(wrapper.find('p').html().includes('- Not Complete -')).toEqual(true);
});

test('should have complete class on todo paragraph tag', () => {
    const newWrapper = shallow(<TodoItem todo={todos[1]} deleteTodo={deleteTodoSpy} toggleCompleted={toggleCompletedSpy} />);
    expect(newWrapper.find('p').hasClass('complete')).toEqual(true);
    expect(newWrapper.find('p').html().includes('- Completed -')).toEqual(true);
});