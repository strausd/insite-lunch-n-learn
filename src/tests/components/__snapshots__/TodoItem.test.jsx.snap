// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TodoItem component properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoItem
    deleteTodo={[MockFunction]}
    todo={
      Object {
        "completed": false,
        "createdAt": 1518030833146,
        "id": "7e53b4e2-5727-482b-8b7f-a851441b8a8b",
        "name": "Finish BBU",
      }
    }
    toggleCompleted={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="not-complete"
        >
          Finish BBU
           - 
          Not Complete
           - 
          02/07/18
        </p>,
        <button
          onClick={[Function]}
        >
          Delete
        </button>,
        <button
          onClick={[Function]}
        >
          Toggle Complete
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Finish BBU",
            " - ",
            "Not Complete",
            " - ",
            "02/07/18",
          ],
          "className": "not-complete",
        },
        "ref": null,
        "rendered": Array [
          "Finish BBU",
          " - ",
          "Not Complete",
          " - ",
          "02/07/18",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Delete",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Delete",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Toggle Complete",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Toggle Complete",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="not-complete"
          >
            Finish BBU
             - 
            Not Complete
             - 
            02/07/18
          </p>,
          <button
            onClick={[Function]}
          >
            Delete
          </button>,
          <button
            onClick={[Function]}
          >
            Toggle Complete
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Finish BBU",
              " - ",
              "Not Complete",
              " - ",
              "02/07/18",
            ],
            "className": "not-complete",
          },
          "ref": null,
          "rendered": Array [
            "Finish BBU",
            " - ",
            "Not Complete",
            " - ",
            "02/07/18",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Toggle Complete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Toggle Complete",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
